app/lib/features/chat_ng/widgets/chat_messages.dart

```diff
@@ -36,11 +38,9 @@ class _ChatMessagesConsumerState extends ConsumerState<ChatMessages> {
     ref.listenManual(
       chatMessagesStateProvider(
         widget.roomId,
-      ).select((value) => value.messageList.length),
-      (_, __) {
-        if (_scrollController.hasClients &&
-            _scrollController.position.pixels >
-                _scrollController.position.maxScrollExtent - 150) {
+      ).select((value) => value.messageList),
```

Comment:
I know this doesn't sound easy, but I think especially now that we are growing its complexity, this widget _needs tests_ for all these complex problems. all these complex problems that are being tried to address are really hard to recreate in real life and even harder to follow in any review. this needs tests.